File: app/src/main/java/com/OdiousPanda/rainbow/API/WeatherCall.java
package com.OdiousPanda.rainbow.API;

import com.OdiousPanda.rainbow.DataModel.Weather.Weather;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface WeatherCall {
    @GET(APIConstants.API_KEY + "/{lat},{lon}")
    Call<Weather> getWeather(@Path("lat") String lat, @Path("lon") String lon, @Query("lang") String lang);
}

File: app/src/main/java/com/odiousPanda/rainbowKt/apis/WeatherCall.kt
package com.odiousPanda.rainbowKt.apis

import com.odiousPanda.rainbowKt.model.dataSource.weather.Weather
import retrofit2.Response

import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

interface WeatherCall {
    @GET(APIConstants.API_KEY + "/{lat},{lon}")
    suspend fun getWeather(
        @Path("lat") lat: String,
        @Path("lon") lon: String,
        @Query("lang") lang: String?
    ): Response<Weather>
}